### PR TITLE
[Tool] Add nightly failure classification artifact

### LINKED ISSUE #739: [Harness] Produce structured nightly failure classification
## Context

The Feishu nightly report has been simplified, but nightly failures are still not emitted as a standard structured artifact.

Without a machine-readable classification, failures remain harder to route and trend over time.

## Proposed classification

Emit a JSON summary for each nightly run with categories such as:

- product_regression
- provider_outage
- docker_infra
- known_flaky
- unknown_flaky
- harness_setup
- credential_or_config

The Feishu message can stay concise, but it should link to or summarize this structured artifact.

## Acceptance criteria

- Nightly uploads a JSON classification artifact even on failure.
- Feishu notification uses the structured classification instead of raw log snippets where possible.
- Known flaky / registered warning handling is represented explicitly.
- Unknown failures are preserved with enough log pointers for triage.

### PR DESCRIPTION
**Pull request by stukid**
## Summary
- Generate `nightly-failure-classification.json` from the nightly manifest, nightly log, and flaky registry.
- Classify failures into product regression, provider outage, docker infra, known flaky, unknown flaky, harness setup, credential/config, or unknown failure.
- Upload the classification artifact and use it in the Feishu nightly report before falling back to raw log snippets.

## Validation
- `uv run ruff format --check ci/classify_nightly_failures.py tests/unit_tests/ci/test_classify_nightly_failures.py`
- `uv run ruff check ci/classify_nightly_failures.py tests/unit_tests/ci/test_classify_nightly_failures.py`
- `uv run pytest tests/unit_tests/ci/test_classify_nightly_failures.py tests/unit_tests/ci/test_check_flaky_registry.py -q`
- `bash -n ci/run-nightly-tests.sh`
- `uv run python ci/audit_tests.py --repo-root . --diff-only upstream/main --json /tmp/datus-audit-739-diff.json`
- `TEST_CMD_TIMEOUT=600 uv run python ci/run-pr-tests.py upstream/main`

Closes #739

## Summary by CodeRabbit

* **New Features**
  * Nightly test failures are automatically classified into structured categories (provider outages, service readiness, credential/config issues) and the classification JSON is persisted as a nightly artifact.
  * Classification summaries are incorporated into the nightly report output for clearer visibility of blocking vs non-blocking issues.

* **Chores (CI)**
  * Nightly workflow and runner finalized to produce and export the classification artifact and expose its path.

* **Tests**
  * Added unit tests covering classification behaviors, known-flaky detection, and edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/Datus-ai/Datus-agent/pull/792?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: ci/classify_nightly_failures.py [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+
+
+def read_json(path: Path | None) -> tuple[dict[str, Any] | None, str]:
+    if not path or not path.exists():
+        return None, "missing"
+    try:
+        return json.loads(path.read_text(encoding="utf-8")), ""
+    except json.JSONDecodeError as exc:
+        return None, f"invalid json: {exc}"
+
```
[the remark this answers is not in the document]
Comment (stukid) [reply ci/classify_nightly_failures.py:69]: Fixed. read_json now rejects valid non-object JSON payloads with a harness_setup finding instead of letting later manifest handling crash, and the case is covered by test_non_object_manifest_is_harness_setup_failure.

File: ci/classify_nightly_failures.py [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+                source="manifest.collection",
+                details={
+                    "suite": name,
+                    "collection_exit_code": collection.get("exit_code"),
+                    "raw_tail": collection.get("raw_tail") or [],
+                },
+            )
+
+        if status in {"passed", "skipped"}:
+            continue
```
[the remark this answers is not in the document]
Comment (stukid) [reply ci/classify_nightly_failures.py:220]: Fixed. collection failures now emit the harness_setup collection finding and skip the duplicate suite-failure finding for that suite. Added test_collection_failure_does_not_duplicate_suite_failure.

File: ci/classify_nightly_failures.py [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+
+    if name in {"Flaky Registry Check", "Flaky Log Classification"}:
+        return "harness_setup"
+    if match_any(PROVIDER_SUITE_PATTERNS, text):
+        return "provider_outage"
+    if kind == "compose" or match_any(DOCKER_SUITE_PATTERNS, text):
+        return "docker_infra"
+    if match_any(CREDENTIAL_PATTERNS, log_text):
+        return "credential_or_config"
+    return "product_regression"
```
[the remark this answers is not in the document]
Comment (stukid) [reply ci/classify_nightly_failures.py]: Fixed. suite_category no longer scans the entire nightly log for credential patterns. Credential/config detection now happens only as a global log-pattern finding, and test_verbose_token_test_names_do_not_create_credential_finding covers the verbose pytest filename false-positive case.
